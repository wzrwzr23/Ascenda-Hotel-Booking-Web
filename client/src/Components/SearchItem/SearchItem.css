.searchItem {
    border: 1px solid lightgray;
    padding: 10px;
    border-radius: 5px;
    display: flex;
    justify-content: space-evenly;
    gap: 20px;
    margin-bottom: 20px;
    height: fit-content;
    width: 650px;
  }

  .siAddress{
    font-size: 15px;
    color: black;
  }

  .siCity{
    color: black;
  }

  .siPrice{
    color: black;
  }
  
  .siImg {
    width: 220px;
    height: 220px;
    object-fit: cover;
  }
  
  .siDesc {
    display: flex;
    flex-direction: column;
    gap: 10px;
    flex: 2;
    color: black;
  }
  
  .siTitle {
    font-size: 20px;
    color: #0071c2;
    padding-bottom: 0;
  }
  
  .siTaxiOp {
    font-size: 12px;
    background-color: #008009;
    color: white;
    width: max-content;
    padding: 3px;
    border-radius: 5px;
  }
  
  .siSubtitle{
    font-size: 12px;
    font-weight: bold;
    color: black;
  }
  
  .siFeatures{
    font-size: 12px;
    color: black;
  }
  
  .siCancelOp{
    font-size: 12px;
    color: #008009;
    font-weight: bold;
  }
  
  .siCancelOpSubtitle{
    font-size: 12px;
    color: #008009;
  }
  
  .siDetails {
    flex: 1;
    display: flex;
    flex-direction: column;
    justify-content: space-between
  }
  
  .siRating{
    display: flex;
    width: 30px;
    justify-content: space-between;
    align-self: flex-end;
  }
  
  .siRating>span{
    font-weight: 500;
  }
  .siRating>button{
   background-color: #003580;
   color: white;
   padding: 5px;
   font-weight: bold;
   border: none;
  }
  
  .siDetailTexts{
    text-align: right;
    display: flex;
    flex-direction: column;
    gap: 5px;
  }
  
  .siPrice{
    font-size: 24px;
  }
  
  .siTaxOp{
    font-size: 12px;
    color: gray;
  }
  
  .siCheckButton{
    background-color: #0071c2;
    color: white;
    font-weight: bold;
    padding: 10px 5px;
    border: none;
    cursor: pointer;
    border-radius: 5px;
  }