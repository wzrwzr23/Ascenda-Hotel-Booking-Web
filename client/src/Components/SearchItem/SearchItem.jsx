import './SearchItem.css';
import { Link } from 'react-router-dom';
import React from 'react';

const SearchItem = ({item, price, id}) => {
  return (
    <div className="searchItem" id="searchItem">
        <img className="siImg"  src={(item == undefined ? null : `${item.image_details.prefix}${item.default_image_index}${item.image_details.suffix}`)}  alt={"Hotel Pic"}/>
        <div className="siDesc">
            <h1 className="siTitle">{(item == undefined ? null :item.name)}</h1>
            <div className="siAddress">{(item == undefined ? null :item.address)}</div>
            <div className="siCity">{(item == undefined ? null :item.original_metadata.city)}, {(item == undefined ? null :item.original_metadata.country)}</div>
            <div className="siTaxiOp">Airport taxi with 10% off</div>
            <div className="siSubtitle">
            {(item == undefined ? null :item.title)}
            </div>
            <div className="siCancelOp">Cancellation  with refund!</div>
            <div className="siCancelOpSubtitle">
            You can cancel later, so lock in this great price today!
            </div>
        </div>
        <div className="siDetails">
            <div className="siRating">
            <button>{(item == undefined ? null :item.rating)}</button>
            </div>
            <div className="siDetailTexts">
            <div className="siPrice">S${price}</div>
            <div className="siTaxOp">Includes taxes and fees</div>
            <Link to={/hotelsearch/+id+/roomsearch/+(item == undefined ? null :item.id)}>
              <button className="siCheckButton">See availability</button>
            </Link>
            </div>
        </div>
    </div>
  )
}

export default SearchItem