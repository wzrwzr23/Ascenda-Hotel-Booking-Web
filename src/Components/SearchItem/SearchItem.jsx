import './SearchItem.css';
import { useNavigate } from "react-router-dom";

const SearchItem = () => {
  const navigate = useNavigate();
  const id = 233;
  const handleClicked = () => {
    navigate(`/hotelsearch/id`, { state: { id } });
  };
  return (
    <div className="searchItem">
        <img src="https://pix10.agoda.net/hotelImages/18391689/0/2c6de0f77a916b78928c57f088f08fc6.jpg?ca=19&ce=1&s=1024x768" className="siImg"  alt={"Hotel Pic"}/>
        <div className="siDesc">
            <h1 className="siTitle">Oryx Rotana Hotel</h1>
            <span className="siDistance">500m from center</span>
            <span className="siTaxiOp">Airport taxi with 10% off</span>
            <span className="siSubtitle">
            Hotel with Air conditioning
            </span>
            <span className="siFeatures">
            Entire studio • 1 bathroom • 21m² 1 full bed
            </span>
            <span className="siCancelOp">Cancellation  with refund!</span>
            <span className="siCancelOpSubtitle">
            You can cancel later, so lock in this great price today!
            </span>
        </div>
        <div className="siDetails">
            <div className="siRating">
            <span>Excellent</span>
            <button>8.9</button>
            </div>
            <div className="siDetailTexts">
            <span className="siPrice">S$119</span>
            <span className="siTaxOp">Includes taxes and fees</span>
            <button className="siCheckButton" onClick={handleClicked}>See availability</button>
            </div>
        </div>
    </div>
  )
}

export default SearchItem