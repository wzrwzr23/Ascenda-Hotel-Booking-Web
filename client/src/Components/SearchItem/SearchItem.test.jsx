import SearchItem from './SearchItem'
import React from 'react'
import ReactDOM from 'react-dom'
import { render, screen, cleanup } from '@testing-library/react';
import {jest} from '@testing-library/jest-dom'
import renderer from "react-test-renderer"

afterEach(cleanup);

// it("renders without crashing", ()=>{
//     const div = document.createElement("div")
//     render(
//         <SearchItem>
//         </SearchItem>
//     , div)
//     ReactDOM.unmountComponentAtNode(div)
// })

// it("renders and shows Book Now! button", ()=>{
//     render(<SearchItem></SearchItem>);
//     expect(screen.getByRole('button', {name: /See availability/i, hidden: true})).toBeEnabled();
//     // expect(screen.getByText(/See availability/i, {selector: 'button'}));
// })