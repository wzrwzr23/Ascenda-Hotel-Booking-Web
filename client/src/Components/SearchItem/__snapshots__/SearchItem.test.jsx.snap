// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="searchItem"
  id="searchItem"
>
  <img
    alt="Hotel Pic"
    className="siImg"
    src={null}
  />
  <div
    className="siDesc"
  >
    <h1
      className="siTitle"
    />
    <div
      className="siAddress"
    />
    <div
      className="siCity"
    >
      , 
    </div>
    <div
      className="siTaxiOp"
    >
      Airport taxi with 10% off
    </div>
    <div
      className="siSubtitle"
    />
    <div
      className="siCancelOp"
    >
      Cancellation  with refund!
    </div>
    <div
      className="siCancelOpSubtitle"
    >
      You can cancel later, so lock in this great price today!
    </div>
  </div>
  <div
    className="siDetails"
  >
    <div
      className="siRating"
    >
      <button />
    </div>
    <div
      className="siDetailTexts"
    >
      <div
        className="siPrice"
      >
        S$
      </div>
      <div
        className="siTaxOp"
      >
        Includes taxes and fees
      </div>
      <a
        href="/hotelsearch/undefined/roomsearch/null"
        onClick={[Function]}
      >
        <button
          className="siCheckButton"
        >
          See availability
        </button>
      </a>
    </div>
  </div>
</div>
`;
